plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/validation/ValidationPreferences.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Martin Erich Jobst
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Martin Jobst - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.model.validation;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.preferences.IScopeContext;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.emf.common.util.Diagnostic;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+
+public final class ValidationPreferences {
+
+	public static final String QUALIFIER = "org.eclipse.fordiac.ide.model.validation"; //$NON-NLS-1$
+
+	public static final String HAS_PROJECT_SPECIFIC_OPTIONS = "hasProjectSpecificOptions"; //$NON-NLS-1$
+
+	public static final String SEVERITY_IGNORE = "ignore"; //$NON-NLS-1$
+	public static final String SEVERITY_INFO = "info"; //$NON-NLS-1$
+	public static final String SEVERITY_WARNING = "warning"; //$NON-NLS-1$
+	public static final String SEVERITY_ERROR = "error"; //$NON-NLS-1$
+
+	public static final String COLLISION_SEVERITY = "collisionSeverity"; //$NON-NLS-1$
+
+	public static int getDiagnosticSeverity(final String key, final int defaultValue, final EObject context) {
+		return getDiagnosticSeverity(key, defaultValue, context != null ? context.eResource() : null);
+	}
+
+	public static int getDiagnosticSeverity(final String key, final int defaultValue, final Resource resource) {
+		return getDiagnosticSeverity(key, defaultValue, resource != null ? resource.getURI() : null);
+	}
+
+	public static int getDiagnosticSeverity(final String key, final int defaultValue, final URI uri) {
+		return getDiagnosticSeverity(key, defaultValue,
+				uri != null && uri.isPlatformResource()
```

Comment:
please extract to a local variable. this is hard to read.

Reply:
Done